tracer/src/Datadog.Trace/Propagators/SpanContextPropagator.cs

```diff
@@ -165,9 +166,15 @@ public void Inject<TCarrier>(SpanContext context, TCarrier carrier, Action<TCarr
 
                     if (localSpanContext is not null && spanContext is not null)
                     {
-                        if (localSpanContext.RawTraceId == spanContext.RawTraceId)
+                        if (localSpanContext.TraceId128 == spanContext.TraceId128 || localSpanContext.TraceId128.Lower == spanContext.TraceId128.Lower)
```

Comment:
Why this change? We shouldn't need to compare the `TraceId128.Lower` separately.

Reply:
I got confused regarding the upper vs lower bits of the ids but rolled it back: https://github.com/DataDog/dd-trace-dotnet/pull/5449/commits/bb420fab599dd83a326121b312d8df47261157a0